lib/graphql/execution/interpreter/runtime.rb

```diff
@@ -20,9 +18,18 @@ module GraphQLResult
           attr_accessor :graphql_non_null_field_names
           # @return [nil, true]
           attr_accessor :graphql_non_null_list_items
+
+          # @return [Hash] Plain-Ruby result data (`@graphql_metadata` contains Result wrapper objects)
+          attr_accessor :graphql_result_data
         end
 
-        class GraphQLResultHash < Hash
+        class GraphQLResultHash
+          def initialize
+            # Jump through some hoops to avoid creating this duplicate hash if at all possible.
+            @graphql_metadata = nil
```

Comment:
🤔 what _is_ this for, all the actual metadata is still in attr_accessors via `include GraphQLResult`?

Reply:
I _thought_ ruby would give me a warning if I referenced an instance variable without initializing it first. But I ran the tests with warnings on, and commented out that line, and didn't get a warning. So, no reason 😊  I'll remove it.

Reply:
No I meant, "what is the ivar for at all"? Not "why is it initialized here"?

Reply:
Oh, it's for when this hash contains other `GraphQLResultHash`es. In that case, for the given `key`: 

- `@graphql_metadata[key]` points to the `GraphQLResultHash`
- `@graphql_result_data[key]` points to the `GraphQLResultHash#graphql_result_data`

That way, to return an object of plain Ruby `Hash`es and `Array`s, we only need to call `@response.graphql_result_data` and we get a data structure that's plain Ruby all the way down -- no need for a second traversal.